lib/Actor/include/Actor/Message.h

```diff
@@ -43,33 +45,48 @@ struct MessagePayload : MessagePayloadBase {
   MessagePayload(Args&&... args) : payload(std::forward<Args>(args)...) {}
 
   Payload payload;
+
+  template<typename Inspector>
+  friend inline auto inspect(Inspector& f, MessagePayload& x) {
+    return f.object(x).fields(f.field("payload", x.payload));
+  }
 };
-template<typename Payload, typename Inspector>
-auto inspect(Inspector& f, MessagePayload<Payload>& x) {
-  return f.object(x).fields(f.field("payload", x.payload));
-}
 
 namespace message {
 
+template<typename PID>
+struct ActorDown {
+  PID actor;
+  Error reason;
+
+  template<typename Inspector>
+  friend inline auto inspect(Inspector& f, ActorDown& x) {
+    return f.object(x).fields(f.field("actor", x.actor),
+                              f.field("reason", x.reason));
+  }
+};
+
 template<typename PID>
 struct UnknownMessage {
   PID sender;
   PID receiver;
+
+  template<typename Inspector>
+  friend inline auto inspect(Inspector& f, UnknownMessage& x) {
+    return f.object(x).fields(f.field("sender", x.sender),
+                              f.field("receiver", x.receiver));
+  }
 };
-template<typename Inspector, typename PID>
-auto inspect(Inspector& f, UnknownMessage<PID>& x) {
-  return f.object(x).fields(f.field("sender", x.sender),
-                            f.field("receiver", x.receiver));
-}
```

Comment:
I usually do it like that because it makes a few things easier - we can omit the template parameter, and we can also access private members (though that is not relevant in this case).